CodeEditModules/Modules/Git/src/Client/Live.swift

```diff
@@ -59,6 +59,33 @@ public extension GitClient {
             }
         }
 
+        /// Displays paths that have differences between the index file and the current HEAD commit,
+        /// paths that have differences between the working tree and the index file, and paths in the working tree
+        func getChangedFiles() throws -> [ChangedFiles] {
+            let output = try shellClient.run(
+                "cd \(directoryURL.relativePath.escapedWhiteSpaces());git status -s --porcelain -u"
+            )
+            if output.contains("fatal: not a git repository") {
+                throw GitClientError.notGitRepository
+            }
+            return try output
+                .split(whereSeparator: \.isNewline)
+                .compactMap { line -> ChangedFiles in
```

Comment:
it seems we are not returning nil anywhere else which means we don't have to use a compactMap

Reply:
Imma need you to decide now if you want this to be a map or compactMap...

Reply:
I gave you options. I suggested you to return a nil and use a compactMap but you decided to handle the optional and throw an error, in that case a `compactMap` is not needed. Here it is a good resource for this: 
https://www.hackingwithswift.com/articles/205/whats-the-difference-between-map-flatmap-and-compactmap